### PR TITLE
Closes #1604, added simple OVERLAPS support

### LINKED ISSUE #1604: Missing support for SQL operator 'OVERLAPS'
In SQL it is possible to do something like this:

`select 'yes' from dual where (sysdate-5,sysdate) overlaps (sysdate-2,sysdate-1)` (interval5.sql)

JsqlParser currently does not support this. 

I started a Discussion on how it could be implemented:  https://github.com/JSQLParser/JSqlParser/discussions/1600

### PR DESCRIPTION
**Pull request by d2a-raudenaerde**
Create simple support for OVERLAPS with RowValueConstructor (1 out of 3 cases)

### CODE REVIEW

File: src/main/java/net/sf/jsqlparser/expression/RowTypeConstructor.java [hunk 1]
```diff
@@ -14,14 +14,11 @@
 import net.sf.jsqlparser.parser.ASTNodeAccessImpl;
 import net.sf.jsqlparser.statement.create.table.ColumnDefinition;
 
-public class RowConstructor extends ASTNodeAccessImpl implements Expression {
+public class RowTypeConstructor extends ASTNodeAccessImpl implements Expression {
```
Comment (wumpz) [inline src/main/java/net/sf/jsqlparser/expression/RowTypeConstructor.java:17]: Why did you rename an existing class? This will break code already running on JSqlParser.

File: src/main/java/net/sf/jsqlparser/util/deparser/ExpressionDeParser.java [hunk 2]
```diff
@@ -367,7 +374,7 @@ public void visit(SubSelect subSelect) {
         if (selectVisitor != null) {
             if (subSelect.getWithItemsList() != null) {
                 buffer.append("WITH ");
-                for (Iterator<WithItem> iter = subSelect.getWithItemsList().iterator(); iter.hasNext();) {
+                for (Iterator<WithItem> iter = subSelect.getWithItemsList().iterator(); iter.hasNext(); ) {
```
Comment (wumpz) [inline src/main/java/net/sf/jsqlparser/util/deparser/ExpressionDeParser.java:377]: This reformatting introduced a lot of changes. Reformating should be done in a different PR.

File: src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt [hunk 3]
```diff
@@ -3300,13 +3301,24 @@ Expression Condition():
 {
     [ LOOKAHEAD(2) (<K_NOT> { not=true; } | "!" { not=true; exclamationMarkNot=true; })]
     (
-        LOOKAHEAD(RegularCondition()) result=RegularCondition()
+        LOOKAHEAD(OverlapsCondition()) result=OverlapsCondition()
```
Comment (wumpz) [inline src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt:3304]: Shouldn't this go into SQLCondition?

**Review by wumpz — CHANGES_REQUESTED** [review]

File: src/main/java/net/sf/jsqlparser/expression/RowTypeConstructor.java (hunk 1 above)
Comment (manticore-projects) [reply src/main/java/net/sf/jsqlparser/expression/RowTypeConstructor.java:17]: I second that, please lets not rename Classes without a very strong reason and without any deprecation approach.
It would break code, examples and documentation -- for a rather exotic new Expression.

Comment (manticore-projects) [reply src/main/java/net/sf/jsqlparser/expression/RowTypeConstructor.java:17]:
```sql
select 'yes' from dual where (sysdate-5,sysdate) overlaps (sysdate-2,sysdate-1)
```

Based on the example above, I do see:
```
"(" <EXPRESSION> "," <EXPRESSION> ")" <OVERLAPS> "(" <EXPRESSION> "," <EXPRESSION> ")"
```

Maybe a:
```
<EXPRESSION_LIST> <OVERLAPS> <EXPRESSION_LIST>
```

Please lets not tamper with `RowConstructor` here, it is not a SQL Row at all. Completely different semantics.

Comment (d2a-raudenaerde) [reply src/main/java/net/sf/jsqlparser/expression/RowTypeConstructor.java:17]: Thanks for the feedback, I did not really take into consideration the effect of the rename. I will change it!

Comment (d2a-raudenaerde) [reply src/main/java/net/sf/jsqlparser/expression/RowTypeConstructor.java:17]: There is no such thing as an 'SQL Row' in the sql grammar? They explicitly have two 'types' of row-constructs, one that defines a row-type (used in casts etc), and one that defines values.

File: src/main/java/net/sf/jsqlparser/util/deparser/ExpressionDeParser.java (hunk 2 above)
Comment (d2a-raudenaerde) [reply src/main/java/net/sf/jsqlparser/util/deparser/ExpressionDeParser.java:377]: Agreed.

File: src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt (hunk 3 above)
Comment (d2a-raudenaerde) [reply src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt:3304]: That is a better place, yes.

Comment (d2a-raudenaerde) [reply src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt:3304]: I'll close this PR and open a new one
